src/Arrays/MPIStateArrays.jl

```diff
@@ -169,6 +174,32 @@ function Base.getproperty(Q::MPIStateArray{FT, V}, sym::Symbol) where {FT, V}
     end
 end
 
+"""
+    getstateview(Q, fieldname)
+
+Returns of real element view of the state `fieldname`. The MPI state array `Q`
+must have been generated in a `Vars`-aware fashion. This is mainly to allow accessing fields which line inside of the nested `Vars` state.
+
+The state specified by 'fieldname' can be a 'String', 'Symbol', or 'Expr'
+
+# Examples
+```julia-repl
+julia> S = @vars(x::Float64, y::@vars(α::Float64, β::SVector{3, Float64}))
+julia> Q = MPIStateArray{Float64, S}(MPI.COMM_WORLD, Array, 2, 5, 3)
+julia> β = MPIStateArrays.getstateview(Q, "y.β")
+julia> β = MPIStateArrays.getstateview(Q, :(y.β))
+julia> x = MPIStateArrays.getstateview(Q, :x)
+```
+"""
+
+function getstateview(
+    Q::MPIStateArray{FT, V},
+    fieldname::Union{String, Symbol, Expr},
+) where {FT, V}
+    varrange = varsindices(V, fieldname)
+    return view(realview(Q), :, varrange[1]:varrange[end], :)
```

Comment:
```suggestion
    fieldname::Union{String, Symbol, Expr, Tuple},
) where {FT, V}
    vi = varsindex(V, fieldname)
    return view(realview(Q), :, vi, :)
```

Can we use `varsindex` instead? It looks like it might be more appropriate (it already returns a range) and supports `NTuple` types.

Reply:
no, since we need to handle strings and expressions, whereas `varsindex` only handles symbols:
```julia-repl
julia> S = @vars(x::Float64, y::@vars(α::Float64, β::SVector{3, Float64}));

julia> varsindex(S, :(y.β))
ERROR: MethodError: no method matching unval(::Expr)
Closest candidates are:
  unval(::Val{i}) where i at /home/jekozdon/codes/ClimateMachine.jl/src/Utilities/VariableTemplates/VariableTemplates.jl:50
Stacktrace:
 [1] varsindex(::Type{NamedTuple{(:x, :y),Tuple{Float64,NamedTuple{(:α, :β),Tuple{Float64,SArray{Tuple{3},Float64,1,3}}}}}}, ::Expr) at /home/jekozdon/codes/ClimateMachine.jl/src/Utilities/VariableTemplates/VariableTemplates.jl:60
 [2] top-level scope at REPL[12]:1

julia> varsindex(S, "y.β")
ERROR: MethodError: no method matching unval(::String)
Closest candidates are:
  unval(::Val{i}) where i at /home/jekozdon/codes/ClimateMachine.jl/src/Utilities/VariableTemplates/VariableTemplates.jl:50
Stacktrace:
 [1] varsindex(::Type{NamedTuple{(:x, :y),Tuple{Float64,NamedTuple{(:α, :β),Tuple{Float64,SArray{Tuple{3},Float64,1,3}}}}}}, ::String) at /home/jekozdon/codes/ClimateMachine.jl/src/Utilities/VariableTemplates/VariableTemplates.jl:60
 [2] top-level scope at REPL[13]:1
```